python_modules/libraries/dagster-dg/dagster_dg/docs/app/components/ComponentSchema.tsx

```diff
@@ -13,12 +21,309 @@ export default function ComponentSchema({schema}: Props) {
   }
 
   if (!json) {
-    return <div className={styles.container}>Invalid schema</div>;
+    return <div>Invalid schema</div>;
+  }
+
+  const jsonSchema: JSONSchema7 = json;
+  const title = jsonSchema.title;
+  const defs = jsonSchema.$defs;
+  const properties = jsonSchema.properties ?? {};
+
+  return <Root title={title} properties={properties} defs={defs} />;
+}
+
+function Root({
+  title,
+  properties,
+  defs,
+}: {
+  title: string | undefined;
+  properties: Record<string, JSONSchema7Definition>;
+  defs: Record<string, JSONSchema7Definition> | undefined;
+}) {
+  return (
+    <div className={styles.schemaContainer}>
+      {title ? <div className={styles.schemaTitle}>{title}</div> : null}
+      {Object.entries(properties).map(([key, property]) => (
+        <Property key={key} name={key} property={property} defs={defs} />
+      ))}
+    </div>
+  );
+}
+
+function Property({
+  name,
+  property,
+  defs,
+}: {
+  name: string;
+  property: JSONSchema7Definition;
+  defs: Record<string, JSONSchema7Definition> | undefined;
+}) {
+  const [expanded, setExpanded] = useState(false);
+
+  const onToggleExpansion = () => {
+    setExpanded((current) => !current);
+  };
+
+  if (property === true || property === false) {
+    return null;
+  }
+
+  const {anyOf, type, description, default: defaultValue, $ref, required, examples} = property;
+
+  const expandable =
+    !!$ref ||
+    (type === 'array' &&
+      property.items &&
+      Object.values(property.items).some((item) => typeof item !== 'boolean' && !!item.$ref)) ||
+    !!(
+      anyOf &&
+      anyOf.some(
+        (item) =>
+          typeof item !== 'boolean' &&
+          (item.type === 'object' || item.type === 'array' || !!item.$ref),
+      )
+    );
+
+  const firstExample = examples ? (Array.isArray(examples) ? examples[0] : examples) : null;
+
+  const onClick = () => {
+    console.log(name, property);
```

Comment:
might be a todo?

Reply:
The idea here was that if you click one of these tags, it will expand. Since the tags are potentially pretty complex, though, I don't think I'll keep it.